### PR TITLE
Added Operator Upgrade Integration tests for FMW Domain

### PR DESCRIPTION
**Pull request by hzhao-github**
Added integration tests for Operator upgrade for FMW Domain:

1. Upgrade Operator from 2.6.0 to latest with FMW domain in PV
2. Upgrade Operator from 3.0.0 to latest with FMW domain in PV
3. Upgrade Operator from 3.0.1 to latest with FMW domain in PV
4. Upgrade Operator from 3.0.2 to latest with FMW domain in PV
5. Upgrade Operator from 3.0.3 to latest with FMW domain in PV
6. Upgrade Operator from 3.0.4 to latest with FMW domain in PV
7. Upgrade Operator from 3.1.0 to latest with FMW domain in PV

passed Jenkins job:

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3385 (parallel)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3415 (parallel)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3467 (parallel)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3550 (parallel)

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3494 (sequential)
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3549 (sequential)

### CODE REVIEW

**Review by anpanigr — COMMENTED** [review]
Following runs does not run the upgrade tests

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3372/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3373/

Please start a Upgrade test run in sequential.

**Comment by hzhao-github** [comment]
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3373/parameters/ is a sequential run

**Review by anpanigr — COMMENTED** [review]
Can we extract out the common utility methods ( deployAndAccessApplication, ollectAppAvailability etc)  from ItOperatorUpgrade.java and ItOpUpgradeFmwDomainInPV.java  to a common utility file for easy maintenance.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java [hunk 1]
```diff
@@ -265,7 +261,14 @@ private void upgradeOperator(String operatorVersion, String externalServiceNameS
     if (useHelmUpgrade) {
       // application high availability check only for 3.x releases and later
       // deploy application and access the application once to make sure the app is accessible
-      deployAndAccessApplication(domainNamespace, externalServiceNameSuffix);
+      //deployAndAccessApplication(domainNamespace, externalServiceNameSuffix);
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:264]: remove commented code

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java [hunk 2]
```diff
@@ -176,21 +187,6 @@ public void testOperatorUpgradeFrom3_0_3(@Namespaces(3) List<String> namespaces)
     upgradeOperator("3.0.3", OLD_DEFAULT_EXTERNAL_SERVICE_NAME_SUFFIX, true);
   }
 
-  /**
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:179]: Why are we removing this method ?

**Review by anpanigr — COMMENTED** [review]
Where is the latest sequential  result

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java (hunk 2 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:179]: what are you talking about, it's still there

  @Test
  @DisplayName("Upgrade Operator from 3.0.3 to latest")
  public void testOperatorUpgradeFrom3_0_3(@Namespaces(3) List<String> namespaces) {
    this.namespaces = namespaces;
    upgradeOperator("3.0.3", OLD_DEFAULT_EXTERNAL_SERVICE_NAME_SUFFIX, true);
  }

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java [hunk 3]
```diff
@@ -102,6 +99,19 @@ public static void init() {
 
   }
 
+  /**
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:102]: We do not need upgrade from Release 2.5.0. This test case should be removed.

Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:102]: yes, remove it ..

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java (hunk 2 above)
Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:179]: you are removing testOperatorUpgradeFrom3_0_4, add it back

**Comment by vanajamukkara** [comment]
ItOperatorUpgrade has upgrade tests for operator 3.0.1, 3.0.2, 3.0.3, 3.0.4 to latest. what's the plan for FMW domain tests? Are we going to add them or no?

**Comment by hzhao-github** [comment]
We had discuss about what version to add for the upgrade with FMW domain. The conclusion is to add 2.6.0, 3.0.0 and 3.1.0 for now. But Pani can add more comments

**Review by anpanigr — COMMENTED** [review]
We should do use the same upgrade path(s) in both WLS and FMW images 

See https://build.weblogick8s.org:8443/job/wko-kind-nightly-seqential/91/testReport/oracle.weblogic.kubernetes/ItOperatorUpgrade/

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java (hunk 1 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:264]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java (hunk 2 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:179]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java (hunk 3 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:102]: it's removed in develop branch. I just synced with it

**Review by anpanigr — APPROVED** [review]
LGTM

**Review by vanajamukkara — APPROVED** [review]
copyright needs to be updated, otherwise looks good.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java [hunk 4]
```diff
@@ -0,0 +1,684 @@
+// Copyright (c) 2020, 2021, Oracle Corporation and/or its affiliates.
```
Comment (rjeberhard) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java:1]: "Oracle Corporation" -> "Oracle". Legal changed the required licensing text.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java [hunk 5]
```diff
@@ -39,25 +39,22 @@
 import static oracle.weblogic.kubernetes.TestConstants.RESULTS_ROOT;
 import static oracle.weblogic.kubernetes.TestConstants.WDT_BASIC_IMAGE_NAME;
 import static oracle.weblogic.kubernetes.TestConstants.WDT_BASIC_IMAGE_TAG;
-import static oracle.weblogic.kubernetes.actions.ActionConstants.ITTESTS_DIR;
```
Comment (rjeberhard) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:42]: Fix license year: "2020," becomes "2020, 2021,"

**Review by rjeberhard — CHANGES_REQUESTED** [review]
Please update copyright headers to remove "Corporation" and to add "2021,"

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java (hunk 5 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorUpgrade.java:42]: fixed after sync with develop

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java (hunk 4 above)
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpUpgradeFmwDomainInPV.java]: fixed

**Review by rjeberhard — APPROVED** [review]

**Pull request merged**